services/static-webserver/client/source/class/osparc/desktop/StudyEditorIdlingTracker.js

```diff
@@ -63,16 +64,40 @@ qx.Class.define("osparc.desktop.StudyEditorIdlingTracker", {
     },
 
     __startTimer: function() {
-      this.__idleInterval = setInterval(() => {
+      const checkFn = () => {
+        const timeoutT = osparc.Preferences.getInstance().getUserInactivityThreshold()
+        const warningT = Math.round(timeoutT * 0.8)
         this.__idlingTime++;
-        if (this.__idlingTime >= this.self().IDLE_TIMEOUT) {
-          this.__userIdled();
-        } else if (this.__idlingTime >= this.self().IDLE_WARNING) {
-          this.__updateFlashMessage(this.self().IDLE_TIMEOUT - this.__idlingTime);
+        if (this.__idlingTime >= timeoutT) {
+          if (!this.__frontendTimedout) {
+            // User was inactive in oSPARC, we can test services (ask every minute until response is true or we detect activity again)
+            this.__removeIdleFlashMessage();
+            const checkInactivity = () => {
+              osparc.data.Resources.fetch("studies", "getInactivity", {
+                url: {
+                  studyId: this.__studyUuid
+                }
+              }).then(({data}) => {
+                if (data) {
```

Comment:
Actually inside the data, it provides an "is_inactive" flag that you are supposed to check as well

Reply:
As soon as `is_inactive` is `true` you can stop checking.

Reply:
👍🏼